internal/api/impl/auth/service.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"errors"
+
+	"github.com/perses/perses/internal/api/interface/v1/user"
+	databaseModel "github.com/perses/perses/internal/api/shared/database/model"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+)
+
+// arbitrate decides if we take the old or new string.
+// Return a boolean saying if the result is different from old value.
+func arbitrate(old, new string) (string, bool) {
+	if len(new) > 0 {
+		return new, old != new
+	}
+	return old, false
+}
+
+// saveProfileInfo build a user spec merging the old spec with a given user profile
+// Return a boolean saying if the result is different from old value.
+func saveProfileInfo(old v1.UserSpec, uInfoProfile externalUserInfoProfile) (v1.UserSpec, bool) {
+	hasChanged := false
+	singleChange := false
+	old.FirstName, singleChange = arbitrate(old.FirstName, uInfoProfile.GivenName)
+	hasChanged = hasChanged || singleChange
```

Comment:
```suggestion
hasChanged = singleChange
```

because reading quite quickly you suppose that `hasChanged` can have a different value than the initial one. I had to double check it was the case or not.
Assigning directly `singleChange` remove this question when reading the code

Reply:
You mean for the first one only, yes I can change this one